versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/extension/NessieDbAdapter.java

```diff
@@ -22,4 +22,33 @@
 
 @Target({ElementType.FIELD, ElementType.PARAMETER})
 @Retention(RetentionPolicy.RUNTIME)
-public @interface NessieDbAdapter {}
+public @interface NessieDbAdapter {
+
+  /**
+   * Optional: name of method to update the {@link
+   * org.projectnessie.versioned.persist.adapter.DatabaseAdapterConfig configuration} for the {@link
+   * org.projectnessie.versioned.persist.adapter.DatabaseAdapter}.
+   *
+   * <p>The method must be
+   *
+   * <ul>
+   *   <li>static
+   *   <li>not private
+   *   <li>have a single parameter {@code
+   *       org.projectnessie.versioned.persist.adapter.DatabaseAdapterConfig}
+   *   <li>return {@code org.projectnessie.versioned.persist.adapter.DatabaseAdapterConfig}
+   * </ul>
+   *
+   * <p>Example:
+   *
+   * <code><pre>
+   *   &#64;NessieDbAdapter(configMethod = "applyTestClock")
+   *   protected static DatabaseAdapter databaseAdapter;
+   *
+   *   static DatabaseAdapterConfig applyTestClock(DatabaseAdapterConfig config) {
+   *     return config.withClock(TEST_CLOCK);
```

Comment:
How about:
```
 static TxDatabaseAdapterConfig applyTestClock(TxDatabaseAdapterConfig config) {
   return ImmutableTxDatabaseAdapterConfig.builder().from(config).clock(TEST_CLOCK).build();
}
```

Reply:
But what if it’s a non-tx Adapter?
Would Quarkus complain about a `toBuilder()` function on the interfaces?

Reply:
but, whoever uses the `NessieDbAdapter` annotation should know the expected adapter and config types, right?

Reply:
... not sure about `toBuilder()`... what is it's class? Quarkus-generated configs should be usage in `builder().from(config)` AFAIK

Reply:
Reopened #2148 - it broke functionality.